lms/djangoapps/courseware/access.py

```diff
@@ -279,9 +279,30 @@ def can_view_courseware_with_prerequisites():  # pylint: disable=invalid-name
         else:
             return True
 
+    def can_view_courseware_with_entrance_exam():  # pylint: disable=invalid-name
+        """
+        Checks if entrance exam feature is enabled and course has entrance exam
+        and user is neither staff nor allowed to skip entrance exam
+        then it returns False if user has not passed entrance exam
+        otherwise returns True. In case of anonymous user returns False also.
+        """
+        if settings.FEATURES.get('ENTRANCE_EXAMS') and getattr(course, 'entrance_exam_enabled', False):
+            if user.is_anonymous():
+                return False
+
+            if not _has_staff_access_to_descriptor(user, course, course.id) \
```

Comment:
Not a big deal, but again here's another place that gets more complicated than if a member of staff could always skip entrance exam.